src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -1592,11 +1675,13 @@ private void GenerateVTableMethodDelegates(Class @class, Method method)
             var vTableMethodDelegateName = GetVTableMethodDelegateName(method);
             WriteLine("internal delegate {0} {1}({2});", retType, vTableMethodDelegateName,
                 string.Join(", ", @params));
+             * */
+            var vTableMethodDelegateName = GenerateDelegate(method);
 
-            WriteLine("private static {0} {0}Instance;", vTableMethodDelegateName);
+            WriteLine("private static {0} {1}Instance;", vTableMethodDelegateName, vTableMethodDelegateName.Substring(17));
             NewLine();
 
-            WriteLine("private static {0} {1}Hook({2})", retType, vTableMethodDelegateName,
+            WriteLine("private static {0} {1}Hook({2})", retType, vTableMethodDelegateName.Substring(17),
```

Comment:
I would guess it's here that you get out of range. Could you please elaborate what this piece is for?
By the way, if you fell it would go faster, feel free to contact me over Skype.